### PR TITLE
feat: github action to check the merge conflict on push

### LINKED ISSUE #181: create global workflow that informs PR author about merge conflicts
#### Reason/Context

Sometimes PRs get merge conflicts. No notifications tell the PR creator about it. This causes some PRs to hang for some time. It affects release branches the most

#### Description

We need workflow, that once it detects merge conflict on a PR, it posts a comment in PR and mentions the author. The comment should inform the author that there is a merge conflict (this will trigger notification).

It should not be annoying and duplicate comments, so we can inspire on https://github.com/asyncapi/.github/pull/175 and:
- put new comments only if non exists so far
- delete comment if merge conflict is resolved

Bonus:
Maybe some tips on what is the best way to solve the merge conflict 🤔 good and bad practices.

Super bonus: 
Additional workflow that anyone can trigger by adding comments like /rebase or /solve_conflict (/sf) would trigger a workflow that tries to solve the conflict in an automated way. Something to explore

### PR DESCRIPTION
**Pull request by sambhavgupta0705**
**Description**

current PR adds a github actions which will check all the open PR when a push commit is made and will leave a comment if there is a merge conflict.

**Related issue(s)**

Resolves #181

### CODE REVIEW

**Comment by sambhavgupta0705** [comment]
hey @derberg will require your lot of help in this one.
I checked it on my local  project first and it was working fine there.
Kindly review it let me know the areas where I can improve.

File: .github/workflows/merge_conflicts.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+on: push
+
+permissions:
+  pull-requests: write
+
+jobs:
+  auto-comment-merge-conflicts:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: codytseng/auto-comment-merge-conflicts@v1
```
Comment (derberg) [inline .github/workflows/merge_conflicts.yaml:12]:
```suggestion
      - uses: codytseng/auto-comment-merge-conflicts@d621e322650866ded2c0753ea565dd83a18e01ff #using 1.3 version https://github.com/CodyTseng/auto-comment-merge-conflicts/releases/tag/v1.3.0
```

File: .github/workflows/merge_conflicts.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: Auto Comment Merge Conflicts
+
+on: push
+
+permissions:
```
Comment (derberg) [inline .github/workflows/merge_conflicts.yaml:5]: why is this needed? we never used it before

File: .github/workflows/merge_conflicts.yaml (hunk 1 above)
Comment (derberg) [reply .github/workflows/merge_conflicts.yaml:12]: we should always try to use commitId from actions that do not belong to GitHub itself, for security reason.

File: .github/workflows/merge_conflicts.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  pull-requests: write
+
+jobs:
+  auto-comment-merge-conflicts:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: codytseng/auto-comment-merge-conflicts@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          comment-body: "Hey there 👋🏼, your current PR has some merge conflicts. KIndly try to resolve them in your local fork and push the new changes."
```
Comment (derberg) [inline .github/workflows/merge_conflicts.yaml:15]: maybe we could give some hints? instructions? or maybe even action could figure that if conflict is with package-lock, what are the options to solve it?

**Comment by KhudaDad414** [comment]
hey @sambhavgupta0705, are you still working on this issue?

**Comment by sambhavgupta0705** [comment]
Yes @KhudaDad414 ,I was just busy with other work and my exams.
I will be resolving it ASAP.
